File: day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {

	file, _ := os.Open("t.txt")
	scanner := bufio.NewScanner(file)

	ans := 50
	cnt := 0

	for scanner.Scan() {
		line := scanner.Text()
		side := line[0]
		dis, _ := strconv.Atoi(line[1:])
		if side == 'L' {
			for i := dis; i > 0; i-- {
				ans--
				ans %= 100
				if ans == 0 {
					cnt++
				}
			}

		} else {
			for i := dis; i > 0; i-- {
				ans++
				ans %= 100
				if ans == 0 {
					cnt++
				}
			}
		}
	}
	fmt.Println(cnt)

}

File: day1/main.py
with open("t.txt", "r") as f:
    parts = f.read().strip().split("\n")
# ans = 289042678603
ans = 50
rnt = 0
for i in parts:
    side = i[0]
    dis = int(i[1:])

    if side == "L":
        for i in range(dis, 0, -1):
            ans -= 1
            ans %= 100
            if ans == 0:
                rnt += 1
    else:
        for i in range(dis, 0, -1):
            ans += 1
            ans %= 100
            if ans == 0:
                rnt += 1


print(rnt)
